keyboards/keychron/q1/rev_0100/keymaps/robertolupi/keymap.c

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2022 Roberto Lupi <roberto.lupi@gmail.com> @robertolupi
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "robertolupi.h"
+
+enum layers {
+    MAC_BASE,
+    MAC_FN,
+    WIN_BASE,
+    WIN_FN,
+};
+
+enum { TD_CAPS };
+
+qk_tap_dance_action_t tap_dance_actions[] = {
+    [TD_CAPS] = ACTION_TAP_DANCE_DOUBLE(CAPS_WORD, KC_CAPS),
```

Comment:
I'm not sure that this works, as I believe the "dance double" uses register_code16, which supports only basic (and modded) keycodes.